### PR TITLE
Initial addition of graphing capabilities

### PR DESCRIPTION
**Pull request by MatzElectronics**
Needed to get this code in before the branch diverged too much further (already had to resolve merge conflicts).

Visible changes are all "double wrapped" (invisible unless deliberately activated).  To enable graphing blocks and modals, add "&graphing=1" to the end of the URL.

Uses specially-formatted (CSV-like) values from the Propeller and graphs them.  Uses a library available through JSDelivr CDN called Chartist.js.

Initial testing shows upwards of 25ms refresh rates on incoming data being graphed.  I expect it to still be a bit buggy, but otherwise functional (hey, can't be too much worse than the terminal, right?)

Addresses #1019

### CODE REVIEW

File: src/main/webapp/editor/blocklyc.jsp [hunk 1]
```diff
@@ -23,6 +23,8 @@
         <meta charset="utf-8">
         <meta name="base" content="<url:getUrl url="/"/>">
         <title>BlocklyProp</title>
+        <script type="text/javascript" src="https://cdn.jsdelivr.net/chartist.js/latest/chartist.min.js"></script>
```
Comment (paragitadmin) [inline src/main/webapp/editor/blocklyc.jsp]: We need to pull this package into the webapp/cdn/lib folder to eliminate the off-site reference.

File: src/main/webapp/editor/blocklyc.jsp [hunk 2]
```diff
@@ -32,6 +34,9 @@
         <script type="text/javascript" src="<url:getCdnUrl url="/blocklyc.js"/>"></script>
         <script type="text/javascript" src="<url:getCdnUrl url="/utils.js"/>"></script>
         <script type="text/javascript" src="<url:getCdnUrl url="/editor.js"/>"></script>
+
+        <link href="https://cdn.jsdelivr.net/chartist.js/latest/chartist.min.css" rel="stylesheet" type="text/css" />
```
Comment (paragitadmin) [inline src/main/webapp/editor/blocklyc.jsp]: Same thing with the CSS file. Please load up the desired version into the webapp/cdn/lib/ folder.

**Comment by MatzElectronics** [comment]
Will do!

*Matthew Matz  |  STEM/Robotics Educator*
Parallax Inc.  |  Direct: 916-625-3019  |  www.parallax.com  |  @M
atzElectronics <http://twitter.com/MatzElectronics>

**Comment by MatzElectronics** [comment]
So far so good!  Connection stability is an issue - if you close out the graph, and then try to reopen it, it doesn't seem to want to come back to life...

**Comment by wstodghill** [comment]
Is it possible for this to work with the S3?

**Comment by MatzElectronics** [comment]
Should be, yes.  It will start in the activity board and flip first, but
eventually it will move to all boards/devices

**Pull request merged**